src/geometry.h

```diff
@@ -66,6 +66,14 @@ struct quat : gml::quat
   }
 };
 
+// Converting from yaw (around Z), pitch (around Y), roll (around X) to quaternions.
```

Comment:
Oh, I see... I just ran into the same problem as in the GUI head-tracker PR earlier.
We have to document this somewhere. Really.
- I refer to yaw pitch roll as Tait-Bryan angles (I thought this is the most "common" definition).
More specifically, the aerospace norm/ nautical angles.
At least for me, an airplane first yaws on ground, then pitches during take-off and then rolls around its forward (nose) axis.
- I'm not sure if I would understand "north" orientation just from reading it. I think less ambiguous is Y-forward, Z-up. That's at least how coordinate systems got communicated to me, when in doubt (always assuming right handed coordinate systems). Did I understand correctly and do you agree?

We should also make sure to use this convention of YPR consistently across all trackers and maybe even the SSR. I'm not sure if this is the case atm. I thought that at least the Razor uses X-forward, Z-up and the YPR definition above.

Reply:
> We have to document this somewhere. Really.

I agree.
My first attempt is there: https://audioscenedescriptionformat.readthedocs.io/en/latest/position-orientation.html

But this can be for sure improved, and we should mention it in the SSR docs somewhere.

> I refer to yaw pitch roll as Tait-Bryan angles (I thought this is the most "common" definition).

Yeah, I prefer those, too. But there are still 6 possible conventions (not considering left-handed systems and not even considering where x, y and z point to in the real world).

I don't really like the aerospace convention because the z axis points down. I don't think that's intuitive in our situation and for our community.

> I'm not sure if I would understand "north" orientation just from reading it. I think less ambiguous is Y-forward, Z-up.

I would hope that neither is ambiguous. Do you think it is?
I agree that north/east/south/west is a secondary way of looking at it, but it might still be helpful for some people.
And "north" is definitely more concise than "positive y axis".
And there is definitely a tradition of naming those things after compass directions, e.g. "ENU" and "NED" (see https://en.wikipedia.org/wiki/Axes_conventions).

I think that both are equally valid ways of describing the same thing.

> We should also make sure to use this convention of YPR consistently across all trackers

For those tracker APIs that provide quaternions, this whole problem doesn't matter (except for left/right-handedness I guess).

For those that provide angles, we just have to figure out the convention used in their API and correctly convert it to `ssr::quat` or `ssr::Rot`.

But each tracker might use their own convention, therefore I don't think it makes sense to provide a general "ypr2quat()" function.

> I thought that at least the Razor uses X-forward, Z-up and the YPR definition above.

Maybe. But that doesn't mean we have to use that convention for anything else.

It should really be an implementation detail of each tracker, the important thing is that we get a quaternion out.

Reply:
In the meantime I've created a new page in the ASDF docs: https://audioscenedescriptionformat.readthedocs.io/en/latest/rotation-matrices.html

Please tell me your suggestions for improvements. I hope there are no ambiguities left.

At some point I'll probably make a similar page using quaternions, but I think the rotation matrix stuff might be easier to understand anyway.

Reply:
I've added a page about quaternions: https://audioscenedescriptionformat.readthedocs.io/en/latest/quaternions.html